src/tools/perf/mad/perftest_mad.c

```diff
@@ -551,10 +551,9 @@ static ucs_status_t perftest_mad_accept(perftest_mad_rte_group_t *rte_group,
     lid = rte_group->dst_port.lid;
     ucs_debug("MAD: accept: remote lid:%d/0x%02x", lid, lid);
 
-    /* Import receive params, preserving passed flags */
-    needed_flags = ctx->params.super.flags & UCX_PERF_TEST_FLAG_ERR_HANDLING;
+    perftest_params_local_backup(&ctx->params, &params_backup);
     memcpy(&ctx->params, buf, sizeof(ctx->params));
-    ctx->params.super.flags |= needed_flags;
+    perftest_params_local_restore(&ctx->params, &params_backup);
```

Comment:
maybe instead of backup/restore there will be a single pertest_params_copy() function. that function will backup, (to local vars), do memcpy, and restore from local vars - all internally. I think it will simplify the usage.
when receiving from socket, receive into a local struct and then use the new function to copy into ctx->params.

Reply:
if I understand correctly - do you propose to implement different behavior for mad and socket rte?
> receive into a local struct and then use the new function to copy into `ctx->params`

Just `memcpy` won't work because some parameters should be overwritten, for example `msg_size_list` (which furthermore allocated separately) but other one like device name should be saved from initial parameters and third type is `flags` which should be combined (OR'ed). IMO it does not look simpler.